src/AppInstallerRepositoryCore/Microsoft/Schema/1_4/DependenciesTable.cpp

```diff
@@ -0,0 +1,384 @@
+			);
+		}
+
+		void InsertManifestDependencies(
+			SQLite::Connection& connection,
+			SQLite::rowid_t manifestRowId,
+			std::vector<Manifest::Dependency>& dependencies)
+		{
+			using namespace SQLite::Builder;
+			using namespace Schema::V1_0;
+
+			std::vector<Utility::NormalizedString> minVersions;
+			std::vector<Utility::NormalizedString> packageIds;
+			std::for_each(
+				dependencies.begin(),
+				dependencies.end(),
+				[&](Manifest::Dependency dep) { minVersions.emplace_back(dep.MinVersion.value().ToString()); }
+			);
+
+			std::for_each(
+				dependencies.begin(),
+				dependencies.end(),
+				[&](Manifest::Dependency dep) { packageIds.emplace_back(dep.Id); }
+			);
+
+			ManifestTable::ExistsById(connection, manifestRowId);
+
+			// SELECT FROM ids where id IN (Example.Id1, Example.Id2, .... )
+			auto idResultSet = SelectIdsByValues(connection, IdTable::TableName(), IdTable::ValueName(), packageIds);
+			std::vector<Utility::NormalizedString> notFoundIds;
+			FindMissing(idResultSet, packageIds, notFoundIds);
+
+			if (notFoundIds.size() > 0)
+			{
+				std::string failedPackages;
+				std::for_each(
+					notFoundIds.begin(),
+					notFoundIds.end(),
+					[&](Utility::NormalizedString& packageId) {failedPackages.append(std::string{ packageId }); });
+				THROW_HR_MSG(APPINSTALLER_CLI_ERROR_MISSING_PACKAGE, "%s", failedPackages.c_str());
+			}
+
+			// SELECT FROM versions where id IN ("1.0.0", "1.1.0" )
+			auto versionsResultSet = SelectIdsByValues(connection, VersionTable::TableName(), VersionTable::ValueName(), minVersions);
+			std::vector<Utility::NormalizedString> notFoundVersion;
+			FindMissing(versionsResultSet, minVersions, notFoundVersion);
+
+			std::for_each(
+				notFoundVersion.begin(),
+				notFoundVersion.end(),
+				[&](Utility::NormalizedString& version) { versionsResultSet[version] = VersionTable::EnsureExists(connection, version); });
+
+			StatementBuilder insertBuilder;
+			insertBuilder.InsertInto(s_DependenciesTable_Table_Name)
+				.Columns({ s_DependenciesTable_Manifest_Column_Name, s_DependenciesTable_MinVersion_Column_Name, s_DependenciesTable_PackageId_Column_Name })
+				.Values(Unbound, Unbound, Unbound);
+			SQLite::Statement insert = insertBuilder.Prepare(connection);
+
+			// TODO: use insert with multiple values, rather than run the query in a loop.
+			for (const auto& dep : dependencies)
+			{
+				insert.Reset();
+				insert.Bind(1, manifestRowId);
+				insert.Bind(2, versionsResultSet[dep.MinVersion.value().ToString()]);
+				insert.Bind(3, idResultSet[dep.Id]);
+
+				insert.Execute(true);
+			}
+		}
+	}
+
+	std::string_view DependenciesTable::TableName()
+	{
+		return s_DependenciesTable_Table_Name;
+	}
+
+	std::map<std::string, SQLite::rowid_t> SelectIdsByValues(const SQLite::Connection& connection, std::string_view tableName, std::string_view columnName, const std::vector<Utility::NormalizedString>& values)
```

Comment:
You ask tables for their values rather than reaching in.  Remove this in favor of calling the existing methods of getting values.